### DISCUSSION TITLE
Intent/scope of individuals.parent column

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tskit#1582] [General]
What is the scope for the individuals.parent column, and could we document this somewhere? For example, should it only contain actual pedigree parents (i.e. mother and father, if they exist as individuals in the tree sequence), or should it, as @jeromekelleher says "be interpreted in the same way as Edge.parent - indicating a topological relationship, which may or may not correspond to a direct organismal parent/child relationship."? 

If the latter, should we also aim for simplification to do something more sophisticated when deleting individuals who are referenced in the parent column? For example, should it iterate up the existing chain of parents until it terminates in individual IDs which are not marked for deletion?

**Comment by bhaller** [comment]
Hmm; I’m not sure I understand all the ramifications of this idea, but Jerome's proposal seems surprising to me.  If individuals.parent could indicate a grandparental relationship, then I guess it could equally well indicate a great-great-great-great-great-grandparental relationship?  Is that really a useful thing to preserve?  To me, it feels like "nodes" and "edges" are about topology, but an "individual" is, well, an individual, and individuals have parents in the literal biological sense, and if an individual's parent is not represented in the tree sequence then it ought to be null.  If someone does want "the most recent individual(s) that are ancestral to this individual", they can figure that out from the topology, right?

Also: an individual (in a typical biparental sexual situation) has two parents, but four grandparents, eight great-grandparents, etc.  If simplification substitutes grandparents in for parents when parents get simplified away, then does the individual now have four "parents"?  If the grandparents get simplified away and the great-grandparents get substituted, does the individual now have eight "parents"?  If one grandparent gets simplified away but the other three remain, does the individual then have seven "parents", with different levels of actual relatedness to the individual?  Etc.  I'm not sure that I really even understand what's being proposed here.  Apologies if I'm being clueless.  :->

**Reply by hyanwong** [reply]
> If someone does want "the most recent individual(s) that are ancestral to this individual", they can figure that out from the topology, right?

Not if there is no genetic inheritance from that individual, I think?

**Reply by bhaller** [reply]
> > If someone does want "the most recent individual(s) that are ancestral to this individual", they can figure that out from the topology, right?
> 
> Not if there is no genetic inheritance from that individual, I think?

True.  OK, if they are strictly interested in the pedigree relationship between a parent and a child, then would it be useful for tskit to say "Well, hey, I don't have this individual's parent around, but here's the individual's great-great-great-grandparent instead"?  I guess in the end this comes down to utility.  If that is the most useful behavior for the typical use case, then yes, we ought to do that.  But it seems surprising to me, and I have trouble thinking of a scenario in which it would be what I would want, I guess.  But then I'm not a researcher who works with pedigrees.  :->

**Reply by hyanwong** [reply]
I can't make up my mind here, quite. But it would be useful to know if this is an immediate parent or some more distant relative, and if we allow grandparents etc, there's no way to know that at the moment.

**Reply by petrelharp** [reply]
I'm with @bhaller here - the `parent` column should be just the *immediate* parents, since that makes the information *more* useful, and there's too many great^n-grand parents to try to put them in. Besides, if we meant "any most recent genetic ancestor individual", then the information would be derivable from the edge table - you'd just get the individual parents by doing
```
for e in ts.edges():
  if e.child == n:
    print(f"a parent is {ts.node(n).individual}")
```

As another data point: in pyslim, I have to go to a great deal of trouble to figure out who is the direct parent of whom; recording this in the individual table is going to make things a lot easier.

**Reply by jeromekelleher** [reply]
I don't have strong opinions here, happy to go with the consensus.

So, assuming we did really mean "parent", are there properties that we could check and enforce? Does it simplify the problem of checking if the genetic topology is consistent with the pedigree in any way?

**Reply by hyanwong** [reply]
I can completely see why it should be just the immediate parents, and this makes the simplification algorithm easier (in fact, the current implementation is then valid, I think). Note, however, that this does lose information, e.g. upon simplification, unless we can think of another way to encode the information that individual 1 is a nth great grandparent of (say) individual 100 when all the intermediate parent individuals in the tree sequence have been lost, and there is no genetic descent from 1 to 100. This could happen if, for example, we run a WF simulation with no recombination, and simplify it down so that we only keep individuals associated with coalescent nodes. Perhaps in this case there is little reason to keep the pedigree information, since we only lose connections that have no genetic component. But it's worth pointing out.

It may also be worth considering what this means for more unusual mating systems such as parthenogenesis, or organisms that have diploid/haploid alternation of generations. Perhaps this is entirely encapsulated in how we define individuals (e.g. if we define a fern simulation as having alternating diploid and haploid individuals, the parent column would alternate between pointing to one and to two parents).

The original reason for this question was because @MomoLangenstein is encoding a coalescent-generated pedigree tree as a tree sequence, without trying to say anything about genomic inheritance in the individuals, and was wondering how to encode the lines of descent. In this (slightly unusual) case, the individuals are uniparental, so I think we can simply think of it as a clonal genetic tree, with the edges defining the lines of descent.

**Reply by minesrebollo** [reply]
> If someone does want "the most recent individual(s) that are ancestral to this individual", they can figure that out from the topology, right?

What if there are parents that are part of the samples as well as their children? When working with breeding populations this is a very often case...

**Reply by hyanwong** [reply]
That's fine. You can have internal samples in a tree sequence. We should probably have a tutorial or something similar about this. Here, for example, node 7 is an internal sample (some sort of ancestor to the other samples) in the 2nd tree: https://tskit.dev/tutorials/viz.html#leaf-sample-isolated-nodes

**Reply by minesrebollo** [reply]
> That's fine. You can have internal samples in a tree sequence. We should probably have a tutorial or something similar about this. Here, for example, node 7 is an internal sample (some sort of ancestor to the other samples) in the 2nd tree: https://tskit.dev/tutorials/viz.html#leaf-sample-isolated-nodes

Thank you, hadn't seen that tutorial! I guess that would be the case for simulated data but for inferring from real data not ...? In tsdate preprint it says that "In tsinfer, samples cannot directly descend from other samples (which is highly unlikely in reality)"

**Reply by hyanwong** [reply]
> Thank you, hadn't seen that tutorial! I guess that would be the case for simulated data but for inferring from real data not ...? In tsdate preprint it says that "In tsinfer, samples cannot directly descend from other samples (which is highly unlikely in reality)"

Yes, it's quite possible to *represent* a sample as an internal node in a tree sequence. But it's very difficult (almost impossible) to infer from genetic data that a particular sampled genome corresponds to a specific ancestor in an inferred ancestry. This is somewhat similar to the point that phylogeneticists make that you can never prove that a fossil is a true ancestor, so you should never show that in your phylogenetic trees.

Of course, if you have another source of data that says a particular sample is an ancestor of another sample, then you could try to incorporate that into the inference, but at the moment tsinfer can't do that. Instead it allows you to look for descent from genomes that are identical (except for singletons etc) from a historically sampled genome.

This is a tsinfer-specific thing, so perhaps we should direct further discussion of this particular aspect of the question to https://github.com/tskit-dev/tsinfer/discussions/?

**Reply by gregorgorjanc** [reply]
> The original reason for this question was because @MomoLangenstein is encoding a coalescent-generated pedigree tree as a tree sequence, without trying to say anything about genomic inheritance in the individuals, and was wondering how to encode the lines of descent. In this (slightly unusual) case, the individuals are uniparental, so I think we can simply think of it as a clonal genetic tree, with the edges defining the lines of descent.

It’s important to recognise that there are two related data structures convoluted in this discussion (I am not fully up to speed with actual implementation though!) - pedigree between individuals and tree sequence of chromosomes/genomes. Pedigree between individuals indicates expected relationships between individuals and while it’s trivial for parents&progeny (in diploids we get 50% DNA from one parent and 50% DNA from another) the fact that we get a random sample of DNA from each parent means that we inherit a random sample of ancestors. Pedigree between individuals is not capturing this inheritance process - it only gives expectations (genealogical vs actual/genetic/genomic relatedness). Tree sequence is capturing the inheritance process by working at a chromosome/genome level, while pedigree works at the individual level.

So, simplifying a tree should not swap parents with their parents (sample’s grandparents) or more distant ancestors just based on pedigree between individuals. But, it could swap parent of a segment of DNA with the grandparent that passed that DNA segment to the parent, but that information is already stored in edges(I think).

**Reply by jeromekelleher** [reply]
Thanks @gregorgorjanc - it seems like we're all in favour of Individual.parents meaning, literally, their actual parents. The next steps I guess would be to document this, and then figure out what the implications are for how it relates to the genetic ancestry (as you say).
